File: training-data/go/power_set.go
package main

import "fmt"

func powerSet(items []int) [][]int {
	result := [][]int{{}}
	for _, item := range items {
		size := len(result)
		for i := 0; i < size; i++ {
			newSubset := append([]int(nil), result[i]...)
			newSubset = append(newSubset, item)
			result = append(result, newSubset)
		}
	}
	return result
}

func main() {
	fmt.Println(powerSet([]int{1, 2, 3}))
}

File: training-data/java/PowerSet.java
import java.util.ArrayList;
import java.util.List;

public class PowerSet {
    public static List<List<Integer>> powerSet(List<Integer> items) {
        List<List<Integer>> result = new ArrayList<>();
        result.add(new ArrayList<>());
        for (int item : items) {
            int size = result.size();
            for (int i = 0; i < size; i++) {
                List<Integer> newSubset = new ArrayList<>(result.get(i));
                newSubset.add(item);
                result.add(newSubset);
            }
        }
        return result;
    }

    public static void main(String[] args) {
        System.out.println(powerSet(List.of(1, 2, 3)));
    }
}
